bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -509,7 +510,7 @@ public void run() {
                             // queue will benefit from this force write - post a marker prior to issuing
                             // the flush so until this marker is encountered we can skip the force write
                             if (enableGroupForceWrites) {
-                                forceWriteRequests.put(createForceWriteRequest(req.logFile, 0, 0, null, false, true));
+                                forceWriteMarkerSent = forceWriteRequests.offer(createForceWriteRequest(req.logFile, 0, 0, null, false, true));
```

Comment:
consider logging and incrementing a counter (for monitoring/alerting/troubleshooting) if forceWriteMarkerSent is false.

Reply:
createForceWriteRequest uses pool; if offer failed we need to recycle the request to avoid leak.